homeassistant/components/aemet/weather_update_coordinator.py

```diff
@@ -0,0 +1,652 @@
+        if station:
+            self._data["station"] = station
+
+        return AemetWeather(
+            self._data["daily"],
+            self._data["hourly"],
+            self._data["station"],
+        )
+
+    def _convert_weather_response(self, weather_response):
+        """Format the weather response correctly."""
+        if not weather_response or not weather_response.hourly:
+            return None
+
+        elaborated = dt_util.parse_datetime(
+            weather_response.hourly[ATTR_DATA][0][AEMET_ATTR_ELABORATED]
+        )
+        now = dt_util.now()
+        hour = now.hour
+
+        # Get current day
+        day = None
+        for cur_day in weather_response.hourly[ATTR_DATA][0][AEMET_ATTR_FORECAST][
+            AEMET_ATTR_DAY
+        ]:
+            cur_day_date = dt_util.parse_datetime(cur_day[AEMET_ATTR_DATE])
+            if now.date() == cur_day_date.date():
+                day = cur_day
+                break
+
+        # Get station data
+        station_data = None
+        if weather_response.station:
+            station_data = weather_response.station[ATTR_DATA][-1]
+
+        condition = None
+        humidity = None
+        pressure = None
+        rain = None
+        rain_prob = None
+        snow = None
+        snow_prob = None
+        station_id = None
+        station_name = None
+        station_timestamp = None
+        storm_prob = None
+        temperature = None
+        temperature_feeling = None
+        town_id = None
+        town_name = None
+        town_timestamp = dt_util.as_utc(elaborated)
+        wind_bearing = None
+        wind_max_speed = None
+        wind_speed = None
+
+        # Get weather values
+        if day:
+            condition = self._get_condition(day, hour)
+            humidity = self._get_humidity(day, hour)
+            rain = self._get_rain(day, hour)
+            rain_prob = self._get_rain_prob(day, hour)
+            snow = self._get_snow(day, hour)
+            snow_prob = self._get_snow_prob(day, hour)
+            station_id = self._get_station_id()
+            station_name = self._get_station_name()
+            storm_prob = self._get_storm_prob(day, hour)
+            temperature = self._get_temperature(day, hour)
+            temperature_feeling = self._get_temperature_feeling(day, hour)
+            town_id = self._get_town_id()
+            town_name = self._get_town_name()
+            wind_bearing = self._get_wind_bearing(day, hour)
+            wind_max_speed = self._get_wind_max_speed(day, hour)
+            wind_speed = self._get_wind_speed(day, hour)
+
+        # Overwrite weather values with closest station data (if present)
+        if station_data:
+            if AEMET_ATTR_STATION_DATE in station_data:
+                station_dt = dt_util.parse_datetime(
+                    station_data[AEMET_ATTR_STATION_DATE] + "Z"
+                )
+                station_timestamp = dt_util.as_utc(station_dt)
```

Comment:
Sensors with device class timestamp should return a state value with an absolute UTC datetime isoformatted string. We seem to be missing the isoformatting?

Reply:
Should I call utcfromtimestamp then?